test/helm/chart-test.sh

```diff
@@ -163,24 +203,36 @@ test_charts() {
 
     [[ $? == 0 ]] && echo -e "✅ ${GREEN}All charts linted successfully${RESET_FMT}"
     c_echo "------------------------------------------------------------------------------------------------------------------------"
+}
 
-    if [[ $LINT_ONLY == false ]]; then
-        c_echo "Installing helm charts and running tests...\n"
 
-        git remote add upstream https://github.com/aws/amazon-ec2-metadata-mock.git &> /dev/null || true
-        git fetch upstream
+install_and_test_charts() {
+    # provision test env
+    if [[ $REUSE_ENV == false ]]; then
+        setup_ct_container
 
-        if [[ $DEBUG == true ]]; then
-            $CT_EXEC ct install --debug
-        else
-            $CT_EXEC ct install
-        fi
-        [[ $? == 0 ]] && echo -e "✅ ${GREEN}All charts installed and tested successfully${RESET_FMT}"
+        # setup env for chart installation
+        mkdir -p $TMP_DIR
+        install_kind
+        create_kind_cluster
     fi
 
-    if [ $DEBUG == true ]; then
-        set +x
+    c_echo "Installing helm charts and running tests for each *-values.yaml configuration in helm/<chart>/ci dir...\n"
+
+    git remote add upstream https://github.com/aws/amazon-ec2-metadata-mock.git &> /dev/null || true
```

Comment:
what's this for?

Reply:
chart-testing determines changes against the upstream/master configured here - 
https://github.com/aws/amazon-ec2-metadata-mock/blob/master/test/helm/ct.yaml#L8
https://github.com/helm/chart-testing/blob/master/doc/ct_install.md

I configured these because turning on --all disables version checking. 
Figured we can tweak this as we go if we want decide to get rid of version checking for example.